book/providers/http.md

```diff
@@ -1,5 +1,141 @@
+use ethers::providers::{Http, Middleware, Provider};
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
+    let rpc_url = "https://mainnet.infura.io/v3/c60b0bb42f8a4c6481ecd229eddaca27";
+    let provider: Provider<Http> = Provider::<Http>::from_str(rpc_url)?;
+
+    let chain_id = provider.get_chainid().await?;
+    let block_number = provider.get_block_number().await?;
+    let tx_pool_content = provider.txpool_content().await?;
+
+    Ok(())
+}
+```
+
+You can also use the provider to interact with smart contracts. The snippet below uses the provider to establish a new instance of a UniswapV2Pool and uses the `get_reserves()` method from the smart contract to fetch the current state of the pool's reserves. 
+
+```rust
+use std::{str::FromStr, sync::Arc};
+
+use ethers::{
+    prelude::abigen,
+    providers::{Http, Provider},
+    types::H160,
+};
+
+abigen!(
+    IUniswapV2Pair,
+    r#"[function getReserves() external view returns (uint112 reserve0, uint112 reserve1, uint32 blockTimestampLast)]"#
+);
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
+    let rpc_url = "https://mainnet.infura.io/v3/c60b0bb42f8a4c6481ecd229eddaca27";
+    let provider: Arc<Provider<Http>> = Arc::new(Provider::<Http>::try_from(rpc_url).unwrap());
+
+    //Initialize a new instance of the Weth/Dai Uniswap V2 pair contract
+    let pair_address = H160::from_str("0xA478c2975Ab1Ea89e8196811F51A7B7Ade33eB11").unwrap();
+    let uniswap_v2_pair = IUniswapV2Pair::new(pair_address, provider);
+
+    //Use the get_reserves() function to fetch the pool reserves
+    let (reserve_0, reserve_1, block_timestamp_last) =
+        uniswap_v2_pair.get_reserves().call().await?;
+
+    Ok(())
+}
+```
+
+This example is a little more complicated, so lets walk through what is going on. The `IUniswapV2Pair` is a struct that is generated from the `abigen!()` macro. The `IUniswapV2Pair::new()` function is used to create a new instance of the contract, taking in an `Address` and an `Arc<M>` as arguments, where `M` is any type that implements the `Middleware` trait. Note that the provider is wrapped in an [`Arc`](https://doc.rust-lang.org/std/sync/struct.Arc.html) when being passed into the `new()` function. 
+
+It is very common to wrap a provider in an `Arc` to share the provider across threads. Lets look at another example where the provider is used asynchronously across two tokio threads. In the next example, a new provider is initialized and used to asynchronously fetch the number of Ommer blocks from the most recent block, as well as the previous block.
 
 ```rust
-{{#include ../../examples/providers/examples/http.rs}}
-```
\ No newline at end of file
+use std::sync::Arc;
+
+use ethers::providers::{Http, Middleware, Provider};
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
+    let rpc_url = "https://mainnet.infura.io/v3/c60b0bb42f8a4c6481ecd229eddaca27";
+    let provider: Arc<Provider<Http>> = Arc::new(Provider::<Http>::try_from(rpc_url).unwrap());
```

Comment:
```suggestion
    let provider = Arc::new(Provider::<Http>::try_from(rpc_url).unwrap());
```

should work without the type here

let's also avoid unwraps everywhere

i also think you can skip the `::<Http>` because try_from(&str) is only implemented for Provider<Http>